scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/WorkerConfiguration.java

```diff
@@ -17,7 +17,7 @@
 
     public final WorkerTransport transport;
     public final WorkerCQL cql;
-    public final TaskAndRawChangeConsumer consumer;
+    public final Consumer consumer;
```

Comment:
I think this is a step into a wrong direction. Previously, TaskAction was depending only on TaskAndRawChangeConsumer and now it also depends on Consumer and RawChangeConsumer. This creates more coupling than necessary. It seems that consumer field could still be TaskAndRawChangeConsumer and just withConsumer overloads could set it appropriately, no?

Reply:
The point of consumer is to avoid the end user to know anything about what type of dispatch the innermost mechanism actually uses. Because all these things are public (fields!) the Consumer is placed as a variant-like opaque wrapper to hide the actual task interface `ConsumerDispatch`.